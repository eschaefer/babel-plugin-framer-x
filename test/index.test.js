const babel = require('@babel/core');
const plugin = require('../src/index.js');

const example1 = `
const React = require("react");
const { PropertyControls, ControlType } = require("framer");

export class Example extends React.Component {
  static defaultProps = { text: "Hello World!" };

  static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" }
  };

  render() {
    return React.createElement("div", null, this.props.text);
  }
}
`;

const example2 = `
import React from "react";
import { PropertyControls, ControlType } from "framer";

export class Example extends React.Component {
  static defaultProps = { text: "Hello World!" };

  static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" }
  };

  render() {
    return React.createElement("div", null, this.props.text);
  }
}
`;

const plugins = [
	plugin,
	'@babel/plugin-syntax-flow',
	'@babel/plugin-syntax-typescript',
	'@babel/plugin-syntax-class-properties'
];

it('works on imports', () => {
	const { code } = babel.transform(example1, {
		plugins
	});
	expect(code).toMatchSnapshot();
});

it('works on requires', () => {
	const { code } = babel.transform(example2, {
		plugins
	});
	expect(code).toMatchSnapshot();
});
