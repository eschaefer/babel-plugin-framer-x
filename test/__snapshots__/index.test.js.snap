// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works on imports 1`] = `
"const React = require(\\"react\\");

export class Example extends React.Component {
  static defaultProps = {
    text: \\"Hello World!\\"
  };

  render() {
    return React.createElement(\\"div\\", null, this.props.text);
  }

}"
`;

exports[`works on requires 1`] = `
"import React from \\"react\\";
export class Example extends React.Component {
  static defaultProps = {
    text: \\"Hello World!\\"
  };

  render() {
    return React.createElement(\\"div\\", null, this.props.text);
  }

}"
`;
